prebid/src/main/java/com/uid2/prebid/UID2Prebid.kt

```diff
@@ -0,0 +1,132 @@
+package com.uid2.prebid
+
+import com.uid2.UID2Manager
+import com.uid2.UID2ManagerState
+import com.uid2.UID2ManagerState.Established
+import com.uid2.UID2ManagerState.Refreshed
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import org.prebid.mobile.ExternalUserId
+import org.prebid.mobile.PrebidMobile
+
+/**
+ * Interface to wrap access to [PrebidMobile]. This is used to improve testability, rather than having the [UID2Prebid]
+ * access it via static methods.
+ */
+internal fun interface PrebidExternalUserIdInteractor {
+    operator fun invoke(ids: List<ExternalUserId>)
+}
+
+/**
+ * Prebid integration that will observe a given [UID2Manager] instance and update Prebid when a new [ExternalUserId] is
+ * available. After creating the instance, a consumer must explicitly call [initialize] for this instance to start
+ * observing changes.
+ */
+public class UID2Prebid internal constructor(
+    private val manager: UID2Manager,
```

Comment:
```suggestion
    private val manager: UID2Manager = UID2Manager.getInstance(),
```

Should we use the default singleton getter in all the constructors? Seems like a nice to have, to avoid people having to provide their own.